src/cli/java/org/commcare/util/screen/MultiSelectEntityScreen.java

```diff
@@ -0,0 +1,152 @@
+package org.commcare.util.screen;
+
+import static org.commcare.session.SessionFrame.STATE_MULTIPLE_DATUM_VAL;
+
+import org.commcare.core.interfaces.VirtualDataInstanceStorage;
+import org.commcare.data.xml.VirtualInstances;
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.session.CommCareSession;
+import org.commcare.suite.model.MultiSelectEntityDatum;
+import org.javarosa.core.model.instance.ExternalDataInstance;
+import org.javarosa.core.model.instance.ExternalDataInstanceSource;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+
+import javax.annotation.Nullable;
+
+/**
+ * Variation of EntityScreen to allow for selection of multiple entities at once
+ */
+public class MultiSelectEntityScreen extends EntityScreen {
+
+    public static final String USE_SELECTED_VALUES = "use_selected_values";
+    private int maxSelectValue = -1;
+
+    private final VirtualDataInstanceStorage virtualDataInstanceStorage;
+    private String storageReferenceId;
+    private ExternalDataInstance selectedValuesInstance;
+
+    public MultiSelectEntityScreen(boolean handleCaseIndex, boolean full,
```

Comment:
It seems like a MultiSelectEntityScreen will never want to `handleCaseIndex` so I think this could be removed and `false` passed to the super constructor

Reply:
I am assuming that for CLI we would want this to be `true` similar to the [EntityScreen](https://github.com/dimagi/commcare-core/blob/formplayer/src/cli/java/org/commcare/util/cli/ApplicationHost.java#L433) ?